### PR TITLE
--dynamic flag for CLI SO-217

### PR DESCRIPTION
**Pull request by XVincentX**
1. Adds the `--dynamic` CLI flag
2. Prints something on the CLI if the dynamic is true
3. Changes the configuration object to be either "false" or a full configuration object (no more hidden defaults)
4. Adds tests

image

### CODE REVIEW

File: packages/cli/src/const/options.ts [hunk 1]
```diff
@@ -15,4 +15,11 @@ export const FLAGS = {
     default: 4010,
     required: true,
   }),
+
+  dynamic: oflags.boolean({
+    char: 'd',
+    description: 'Dynamically generate examples.',
+    default: true,
```
Comment (philsturgeon) [inline packages/cli/src/const/options.ts]:
```suggestion
    default: false,
```

File: packages/cli/src/commands/mock.ts [hunk 2]
```diff
@@ -14,11 +14,15 @@ export default class Server extends Command {
     signaleInteractiveInstance.await('Starting Prism…');
 
     const {
-      flags: { port },
+      flags: { port, dynamic },
       args: { spec },
     } = this.parse(Server);
 
-    const server = createServer(spec, { mock: true });
+    if (dynamic) {
```
Comment (philsturgeon) [inline packages/cli/src/commands/mock.ts]:
```suggestion
    if (true || dynamic) {
```

**Comment by XVincentX** [comment]
I think @chris-miaskowski should have a look here as well before merging this in.

File: packages/http/src/mocker/__tests__/HttpMocker.spec.ts [hunk 3]
```diff
@@ -191,5 +173,48 @@ describe('HttpMocker', () => {
...
+
+          expect(response.body).toBeDefined();
+
+          const allExamples = flatMap(mockResource.responses, res =>
+            flatMap(res.contents, content => content.examples),
+          ).map(x => {
+            if ('value' in x) return x.value;
+          });
+
+          allExamples.forEach(example => expect(response.body).not.toEqual(example));
```
Comment (chris-miaskowski) [inline packages/http/src/mocker/__tests__/HttpMocker.spec.ts]: This test checks that the response is defined and that it's not one of the defined one, but it doesn't tell us anything about the nature of the response. 

Have you thought about treating this like a white box and maybe stubbing the response? It looks like this file is mostly about unit tests, whereas this test is more of an integration test.

Comment (XVincentX) [reply packages/http/src/mocker/__tests__/HttpMocker.spec.ts]: Done in 007e6a6

**Pull request merged**